website/docs/automation/gitops/argocd-codecommit/deploy_application.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "Deploying an application"
+sidebar_position: 25
+---
+
+We have successfully configured Argo CD on our cluster so now we can deploy an application. To demonstrate the difference between a GitOps-based delivery of an application and other methods, we'll migrate the UI component of the sample application which is currently using the `kubectl apply -k` approach to the new Argo CD deployment approach.
+
+First let's remove the existing UI component so we can replace it:
+
+```bash
+$ kubectl delete -k /workspace/manifests/ui
+namespace "ui" deleted
+serviceaccount "ui" deleted
+configmap "ui" deleted
+service "ui" deleted
+deployment.apps "ui" deleted
+```
+
+Next, clone the CodeCommit repository:
+
+```bash
+$ git clone ssh://${GITOPS_IAM_SSH_KEY_ID}@git-codecommit.${AWS_DEFAULT_REGION}.amazonaws.com/v1/repos/${EKS_CLUSTER_NAME}-gitops-argocd ~/environment/gitops-argocd
+$ cd ~/environment/gitops-argocd && git checkout -b main
+Switched to a new branch 'main'
+```
+
+Now, let's get into the cloned repository and start creating our GitOps configuration. Copy the existing kustomize configuration for the UI service:
+
+```bash
+$ mkdir ~/environment/gitops-argocd/apps
+$ cp -R /workspace/manifests/ui ~/environment/gitops-argocd/apps
+```
+
+We'll then need to create a kustomization in the `apps` directory:
+
+```file
+automation/gitops/argocd-codecommit/kustomization.yaml
+```
+
+Copy this file to the Git repository directory:
+
+```bash
+$ cp /workspace/modules/automation/gitops/argocd-codecommit/kustomization.yaml ~/environment/gitops-argocd/apps/kustomization.yaml
+```
+
+You Git directory should now look something like this which you can validate by running `tree ~/environment/gitops-argocd`:
+
+```
+.
+└── apps
+    ├── kustomization.yaml
+    └── ui
+        ├── configMap.yaml
+        ├── deployment.yaml
+        ├── kustomization.yaml
+        ├── namespace.yaml
+        ├── serviceAccount.yaml
+        └── service.yaml
+
+2 directories, 7 files
+```
+
+Finally we can push our configuration to AWS CodeCommit:
+
+```bash
+$ (cd ~/environment/gitops-argocd && \
+git add . && \
+git commit -am "Adding the UI service" && \
+git push --set-upstream origin main)
+```
+
+It will take Argo CD some time to notice the changes in CodeCommit and reconcile. You can use the Argo CD UI to `Refresh` and `Sync` for our new `apps` kustomization to appear.
+
+Get Argo CD UI url and `admin` password
+
+```bash
+$ echo "ArgoCD URL: http://$(kubectl get svc argo-cd-argocd-server -n argocd -o json | jq --raw-output '.status.loadBalancer.ingress[0].hostname')"
+$ echo "ArgoCD admin password: $(kubectl -n argocd get secret argocd-initial-admin-secret -o jsonpath="{.data.password}" | base64 -d)"
+```
+
+:::info
+
+If you need to reset Argo CD `admin` password, the sequence of commands below can be used:
+
+kubectl -n argocd patch secret argocd-secret -p '{"data": {"admin.password": null, "admin.passwordMtime": null}}'
```

Comment:
You should be able to use a ```bash``` section to wrap these commands so they're a bit easier to see.

Reply:
Used `bash` section